src/status_im/ui/screens/chat/message/message.cljs

```diff
@@ -356,17 +356,42 @@
    [react/view (style/delivery-status outgoing)
     [message-delivery-status message]]])
 
-(defn chat-message [{:keys [message-id outgoing group-chat modal? current-public-key content-type content] :as message}]
+(defview animated-message-body [message content]
+  (letsubs [anim-range   (vector 0 1)
+            anim         (animation/create-value (first anim-range))
+            view-size    (reagent/atom nil)]
+    {:component-did-update (fn [_]
+                             (when @view-size
+                               (animation/start
+                                (animation/timing
+                                 anim
+                                 {:toValue         (second anim-range)
+                                  :easing          (.bezier react/easing 0.165 0.84 0.44 1)
+                                  :duration        400
+                                  :useNativeDriver true}))))}
+    (let [on-layout-handler (fn [event]
+                              (when-not @view-size
+                                (let [layout (-> event .-nativeEvent .-layout)
+                                      height (.-height layout)
+                                      width  (.-width layout)]
+                                  (reset! view-size {:height height :width width}))))]
+      [react/animated-view {:on-layout on-layout-handler
+                            :style (style/animated-message-body anim anim-range message @view-size)}
+       [message-body message content]])))
+
+(defn chat-message [{:keys [message-id outgoing group-chat modal? current-public-key content-type content
+                            last? current-chat-start-time timestamp] :as message}]
   [react/view
    [react/touchable-highlight {:on-press      (fn [_]
                                                 (re-frame/dispatch [:chat.ui/set-chat-ui-props {:messages-focused? true}])
                                                 (react/dismiss-keyboard!))
                                :on-long-press #(when (= content-type constants/text-content-type)
                                                  (list-selection/chat-message message-id (:text content) (i18n/label :t/message)))}
     [react/view {:accessibility-label :chat-item}
-     (let [incoming-group (and group-chat (not outgoing))]
-       [message-content message-body (merge message
-                                            {:current-public-key current-public-key
-                                             :group-chat         group-chat
-                                             :modal?             modal?
-                                             :incoming-group     incoming-group})])]]])
+     (let [incoming-group (and group-chat (not outgoing))
+           animated?      (and last? (> timestamp current-chat-start-time))]
```

Comment:
Is not a good idea to rely on comparison of relative time-stamps in distributed systems, could be out of sync due to network time drift, offline inboxed messages, etc.
Better then maintaining last time chat was opened and comparing it with timestamp of message sent to determine it's a new last message, it's better to just set some flag in message data indicating if it was already animated or not.
Upon completing the animation for particular message, the flag would be simply set by event like `:chat.callback/appear-animation-finished`.

Reply:
Thanks for the comment. I will look into the technical solution you suggested.

Reply:
Hi, I have made the changes based on your technical proposal. Please let me know if there is any problem concerning the changes. Thanks.